text.py

```diff
@@ -380,8 +378,9 @@ def __init__(self, initial=None, goal=None, decoder=None):
 
     def actions(self, state):
         # Find the best
-        p, plainchar = max([(self.decoder.P1[c], c)
-                            for c in alphabet if c not in state])
+                            
+        search_list = [c for c in alphabet if c not in state]
+        plainchar = argmax(search_list, key=lambda c: self.decoder.P1[c])
```

Comment:
This is fine, but I wonder which is clearer:
     lambda c: self.decoder.P1[c]
or
     self.decoder.P1.get